test/face/scan/test_scan.py

```diff
@@ -0,0 +1,476 @@
+    """
+    Attempt failing to load the Tesseract OCR executable
+
+    :return:
+    """
+
+    """
+    Perform the action of loading the Tesseract OCR executable
+    """
+    mocker.patch.object(file.FileHandling, "load_tessexec", side_effect=Exception)
+    qtbot.mouseClick(scantest.arti_cnvs_conf, Qt.LeftButton)
+
+    """
+    Confirm if the user interface elements change accordingly
+    """
+    assert isinstance(scantest.dialog, QMessageBox)
+    assert scantest.dialog.icon() == QMessageBox.Information
+    assert scantest.dialog.windowTitle() == "Faulty scanning"
+    assert "Please consider checking your input after ensuring that the proper Tesseract OCR executable has been selected." in scantest.dialog.text()
+    assert scantest.dialog.isVisible()
+
+
+@pytest.mark.parametrize(
+    "expt",
+    [
+        pytest.param(OSError, id="face.scan.rule: Failing of register function with OSError"),
+        pytest.param(TesseractError("abc", "xyz"), id="face.scan.rule: Failing of register function with TesseractError")
+    ]
+)
+def test_scan_register_fail(scantest, qtbot, mocker, expt) -> None:
+    """
+    Attempt failing the register function with OSError and TesseractError
+
+    :return:
+    """
+
+    """
+    Create the tesseract training data
+    """
+    mocker.patch("gi_loadouts.conf.data_prefix", f"{uuid4().hex.upper()[0:8]}-")
+    make_temp_file()
+
+    """
+    Perform the action of loading the artifact information
+    """
+    temp_prm = ""
+    tempfile = "test/static/img/gi-loadouts-ocr-test.webp"
+    savefile = str(Path(gettempdir()) / "gi-loadouts-ocr-test.webp")
+    with open(tempfile, "rb") as src_file:
+        with open(savefile, "wb") as dest_file:
+            dest_file.write(src_file.read())
+    mocker.patch.object(QFileDialog, "getOpenFileName", return_value=(savefile, temp_prm))
+    mocker.patch("gi_loadouts.face.scan.work.image_to_string", side_effect=expt)
+    qtbot.mouseClick(scantest.arti_cnvs_load, Qt.LeftButton)
+
+    """
+    Confirm if the user interface elements change accordingly
+    """
+    def check_label():
+        """
+        Checking the interface elements change asynchronously
+        """
+        assert isinstance(scantest.dialog, QMessageBox)
+        assert scantest.dialog.icon() == QMessageBox.Information
+        assert scantest.dialog.windowTitle() == "Faulty scanning"
+        if isinstance(expt, OSError):
+            assert "Selected executable of Tesseract OCR is unfunctional." in scantest.dialog.text()
+        elif isinstance(expt, TesseractError):
+            assert "Processing failed as either Tesseract OCR executable ceased to function or training data was tampered with." in scantest.dialog.text()
+        assert scantest.dialog.isVisible()
+    qtbot.waitUntil(check_label)
+
+    """
+    Clear the copied file from temp directory
+    """
+    kill_temp_file()
+    if path.exists(savefile):
+        remove(savefile)
```

Comment:
As the `NamedTemporaryFile`'s path would be reasonably randomized, we need not look for whether the path exists.

```suggestion
    remove(savefile)
```